libblkid/src/superblocks/romfs.c

```diff
@@ -19,10 +19,31 @@
 
 struct romfs_super_block {
 	unsigned char	ros_magic[8];
-	uint32_t	ros_dummy1[2];
+	uint32_t	ros_full_size;
+	uint32_t	ros_checksum;
 	unsigned char	ros_volume[16];
 } __attribute__((packed));
 
+static int romfs_verify_csum(blkid_probe pr, const struct blkid_idmag *mag,
+		const struct romfs_super_block *ros)
+{
+	uint32_t csummed_size = min((uint32_t) 512,
+			be32_to_cpu(ros->ros_full_size));
+	if (csummed_size % sizeof(uint32_t) != 0)
+		return 0;
+
```

Comment:
Please, check csummed_size for non-zero.

Reply:
Wouldn't the existing checksum logic work just fine with `ros_full_size == 0`?
If such a filesystem is invalid it seems better to add the check to `probe_romfs`.

Reply:
@karelzak did you see my response above?